inception-project/inception
# #3413 - Support OAuth2 login with browser

**What's in the PR**
- Added migration for pre-authenticated users to realms
- Added upgrade documentation for pre-authenticated users
- Added generated static meta-objects for JPA entity classes
- Renamed realm `<GLOBAL>` realm to `<LOCAL>`
- Added central methods to check if user has no password, if user can change password and if self-service is allowed
- Moved OAuth2Adapter to security module and added a few tests
- Added builder to User entity class
- Deny access to preauth users that are not enabeld, where the realm of the user does not match the expected preauth realm or where the username is not valid
- Deny access to deactivated users on the AcceptInvitePage
- Added tests for username and document name validation
- Add an echo of the error state on the login page if the OAuth mechanism redirects there with the error parameter being set - unfortunately, we do not know which error
- Add all external and local realms to selection lists if they are configured
- Fix displaying of logout to local users
- Refactor access checking on ManageUsersPage a bit
- Add a notice on the ManageUsersPage if a local user has no password and therefore cannot log in
- Add ability to move users between realms

**How to test manually**
* Configure OAuth2 and test it
* Also test if auto-login can be skipped if configured
* Check if preauth login still works
* Try moving users between realms (project realm-bound users cannot be moved)

**Automatic testing**

**Documentation**
